### ISSUE TITLE
Add/Expose the other OpenGL blend options as shader render_mode

### ISSUE THREAD

**Issue by ghsoares** [godotengine/godot-proposals#2621]
### Describe the project you are working on
Trying to experiment with some shader effects

### Describe the problem or limitation you are having in your project
I'm trying to create a fire effect by taking a viewport that renders particles with radial gradient, procedurally calculate normal from this texture and use it as fake 3D effect in a canvas_item shader, this way it should look like multiple fire spheres overlapping each other. The problem is that I want to achieve a depth effect using `max` blend mode, but the current Godot's shader language API doesn't have this mode exposed

### Describe the feature / enhancement and how it helps to overcome the problem or limitation
Viewing the OpenGL Wiki page about blending (https://www.khronos.org/opengl/wiki/Blending), there's other blend modes like `GL_MIN` and `GL_MAX`, so my proposal is to expose these modes into the Godot's shader language.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams
I ain't sure how it's implemented in the rasterizer and in the shader language, but it seens not too hard to expose/add those modes.

### If this enhancement will not be used often, can it be worked around with a few lines of script?
Not possible, it requires modification in the engine rendering core.

### Is there a reason why this should be core and not an add-on in the asset library?
Because it requires modification in the engine rendering core.

**Comment by Calinou** [comment]
Does OpenGL ES expose these blending modes? If not, they'll have to be emulated in a shader, which is slower.

**Comment by ghsoares** [comment]
@Calinou OpenGL ES 3.0 does have trough the `glBlendEquation` https://www.khronos.org/registry/OpenGL-Refpages/es3.0/html/glBlendEquation.xhtml, but version 2.0 don't have min and max blend modes https://www.khronos.org/registry/OpenGL-Refpages/es2.0/xhtml/glBlendEquation.xml

**Comment by LunaticInAHat** [comment]
Should this proposal to be limited to just the Min and Max blend equations? Would we want to also make the full gamut of blend functions available? Right now I believe our options are limited to (in GL terms, although this applies regardless of renderer):
 - `glBlendFuncSeparate(GL_SRC_ALPHA, GL_ONE_MINUS_SRC_ALPHA, GL_ONE, GL_ONE_MINUS_SRC_ALPHA);` (Used by `blend_mix`)
 - `glBlendFuncSeparate(GL_SRC_ALPHA, GL_ONE, GL_SRC_ALPHA, GL_ONE);` (Used by `blend_add` and `blend_sub`)
 - `glBlendFuncSeparate(GL_DST_COLOR, GL_ZERO, GL_DST_ALPHA, GL_ZERO);` (Used by `blend_mul`)
 - `glBlendFuncSeparate(GL_SRC_ALPHA, GL_ONE_MINUS_SRC_ALPHA, GL_ONE, GL_ZERO);` (Used by when AA is enabled)
 - `glBlendFuncSeparate(GL_ONE, GL_ONE_MINUS_SRC_ALPHA, GL_ONE, GL_ONE_MINUS_SRC_ALPHA);` (Used by `blend_premul_alpha`)

Those are surely the most commonly used blend modes, but special effects sometimes like to use obscure blend modes (as an example, World of Warcraft uses a bunch of unusual blend modes, in some of their effects, like `GL_DST_COLOR, GL_SRC_COLOR, GL_DST_ALPHA, GL_SRC_ALPHA`)

I would really appreciate having some way to just directly specify the blend function that I want, without being confined to the five options that most people use.

**Comment by Calinou** [comment]
Are these blend modes also available in WebGL 2.0? Remember that the Compatibility rendering method should be able to run on that subset of OpenGL ES 3.0 (which is a subset of OpenGL 3.3 already).

> I would really appreciate having some way to just directly specify the blend function that I want, without being confined to the five options that most people use.

Out of curiosity, what kind of effects do you need these extra blend modes for?

**Comment by LunaticInAHat** [comment]
> [@LunaticInAHat](https://github.com/LunaticInAHat) Are the blend modes you mentioned also available in WebGL 2.0? Remember that the Compatibility rendering method should be able to run on that subset of OpenGL ES 3.0 (which is a subset of OpenGL 3.3 already).

Yep, they are. Most of these blend modes have been supported since OpenGL 1.x. For the sake of completeness, I looked them up in the [WebGL 1.0 spec](https://registry.khronos.org/webgl/specs/latest/1.0/), and they all appear to be listed (search for `BlendingFactorDest`, `BlendingFactorSrc`, and `Separate Blend Functions`).

> Out of curiosity, what kind of effects do you need these extra blend modes for?

There are also applications like #7058, where the goal is to mimic the look and feel of an older game that used some of the less common blend modes. This is where my own interest comes from; World of Warcraft used a variety of unusual blending modes (see [this table](https://wowdev.wiki/Rendering#EGxBlend), if curious) for drawing spell effects, or fake specular effects, and it's not possible to get models looking the same as they did in WoW, when limited to just Mix, Mul, and Add.

There are also things like [this](https://github.com/MizunagiKB/gd_cubism/issues/7), where the lack of blending control is apparently making it difficult for the artist to reproduce the results they got out of another program (the differences are a bit subtle, to my eye, but the character's hair is quite visibly different)

**Comment by milesturin** [comment]
I also require GL_MIN/GL_MAX for a shader effect: I have overlapping 2d tracks in mud. In order to merge them properly, I need to be able to take the minimum color at each fragment. As far as I can tell any workaround is going to be significantly more expensive and inconvenient.
